547: Test Worksheets (Backend)

teetone opened the pull request:
Fixes #547 

- Use regex to validate the output of `cl print` for the sample worksheet
- Hook it up to `test_cli`

I had to modify some of the example search criteria in order to not add any new dependencies (e.g. search for `ready` runs instead of `failed` runs).

teetone committed:
547: update create worksheet script

teetone committed:
Merge branch 'master' of https://github.com/codalab/codalab-worksheets into 547

teetone committed:
cleanup and refactor

teetone committed:
pass in cl

percyliang reviewed (changes requested):
Great!  Just a comment about naming.

teetone committed:
Merge branch 'master' of https://github.com/codalab/codalab-worksheets into 547

teetone committed:
547: cleanup

percyliang reviewed (approved):
How long does the new test take to run?  Can you attach a sample output?

teetone commented:
@percyliang The new test takes ~1 minute from start to finish. If there are any discrepancies between the output and expected regex, the Travis build will fail and it will output the mismatches.

Output:
```
Namespace(cl_executable='cl', cleanup=False, large=False, preview=False, test_print=True)
Creating a small worksheet...
>> cl new valid_worksheet_6911p0iovbtdrlr6626kqmug4083fv8s
 (exit code 0, expected 0)

0x4768b82eaf7b40b088ae9dd8cbeb5d43

>> cl work valid_worksheet_6911p0iovbtdrlr6626kqmug4083fv8s
 (exit code 0, expected 0)

Switched to worksheet: http://rest-server:2900/worksheets/0x4768b82eaf7b40b088ae9dd8cbeb5d43 (valid_worksheet_6911p0iovbtdrlr6626kqmug4083fv8s)

>> cl wedit --tag=codalab-sample-worksheet --title=Other Worksheet 6911p0iovbtdrlr6626kqmug4083fv8s
 (exit code 0, expected 0)

Saved worksheet metadata for valid_worksheet_6911p0iovbtdrlr6626kqmug4083fv8s(0x4768b82eaf7b40b088ae9dd8cbeb5d43).

>> cl run echo codalab rules! --tags=codalab-sample-worksheet
 (exit code 0, expected 0)

0x5da14fa649d049e782e85ef37f1d345a

>> cl new valid_private_worksheet_6911p0iovbtdrlr6626kqmug4083fv8s
 (exit code 0, expected 0)

0x2d5005a85ca24e0096434cbd7bee59ba

>> cl work valid_private_worksheet_6911p0iovbtdrlr6626kqmug4083fv8s
 (exit code 0, expected 0)

Switched to worksheet: http://rest-server:2900/worksheets/0x2d5005a85ca24e0096434cbd7bee59ba (valid_private_worksheet_6911p0iovbtdrlr6626kqmug4083fv8s)

>> cl wedit --tag=codalab-sample-worksheet --title=Other Private Worksheet 6911p0iovbtdrlr6626kqmug4083fv8s
 (exit code 0, expected 0)

Saved worksheet metadata for valid_private_worksheet_6911p0iovbtdrlr6626kqmug4083fv8s(0x2d5005a85ca24e0096434cbd7bee59ba).

>> cl wperm 0x2d5005a85ca24e0096434cbd7bee59ba public none
 (exit code 0, expected 0)

Group public(0xc573c2c89326443a97e96edaf6443e51) has none permission on worksheet valid_private_worksheet_6911p0iovbtdrlr6626kqmug4083fv8s(0x2d5005a85ca24e0096434cbd7bee59ba).

>> cl run echo private run --tags=codalab-sample-worksheet
 (exit code 0, expected 0)

0x85b0a3242a3d4f90878a97ecf132f85b

>> cl perm 0x85b0a3242a3d4f90878a97ecf132f85b public none
 (exit code 0, expected 0)

Group public(0xc573c2c89326443a97e96edaf6443e51) has none permission on 1 bundles.

>> cl new valid_worksheet_dg72kyu0q0a4hk9l9qbhpukflkae4nfs
 (exit code 0, expected 0)

0x49c42dde0ad74043881cbae26fd90882

>> cl work valid_worksheet_dg72kyu0q0a4hk9l9qbhpukflkae4nfs
 (exit code 0, expected 0)

Switched to worksheet: http://rest-server:2900/worksheets/0x49c42dde0ad74043881cbae26fd90882 (valid_worksheet_dg72kyu0q0a4hk9l9qbhpukflkae4nfs)

>> cl wedit --tag=codalab-sample-worksheet --title=Other Worksheet dg72kyu0q0a4hk9l9qbhpukflkae4nfs
 (exit code 0, expected 0)

Saved worksheet metadata for valid_worksheet_dg72kyu0q0a4hk9l9qbhpukflkae4nfs(0x49c42dde0ad74043881cbae26fd90882).

>> cl run echo codalab rules! --tags=codalab-sample-worksheet
 (exit code 0, expected 0)

0x31a7a1f87d58453e8a839e05b4dd960b

>> cl new valid_private_worksheet_dg72kyu0q0a4hk9l9qbhpukflkae4nfs
 (exit code 0, expected 0)

0xb2a212ef2a7145709e82267e7047f5d7

>> cl work valid_private_worksheet_dg72kyu0q0a4hk9l9qbhpukflkae4nfs
 (exit code 0, expected 0)

Switched to worksheet: http://rest-server:2900/worksheets/0xb2a212ef2a7145709e82267e7047f5d7 (valid_private_worksheet_dg72kyu0q0a4hk9l9qbhpukflkae4nfs)

>> cl wedit --tag=codalab-sample-worksheet --title=Other Private Worksheet dg72kyu0q0a4hk9l9qbhpukflkae4nfs
 (exit code 0, expected 0)

Saved worksheet metadata for valid_private_worksheet_dg72kyu0q0a4hk9l9qbhpukflkae4nfs(0xb2a212ef2a7145709e82267e7047f5d7).

>> cl wperm 0xb2a212ef2a7145709e82267e7047f5d7 public none
 (exit code 0, expected 0)

Group public(0xc573c2c89326443a97e96edaf6443e51) has none permission on worksheet valid_private_worksheet_dg72kyu0q0a4hk9l9qbhpukflkae4nfs(0xb2a212ef2a7145709e82267e7047f5d7).

>> cl run echo private run --tags=codalab-sample-worksheet
 (exit code 0, expected 0)

0x46871f0635c64cfd9a978b6713bb776e

>> cl perm 0x46871f0635c64cfd9a978b6713bb776e public none
 (exit code 0, expected 0)

Group public(0xc573c2c89326443a97e96edaf6443e51) has none permission on 1 bundles.

>> cl new valid_worksheet_q6d4agn39u5uht8i18k9j8v335t98jqo
 (exit code 0, expected 0)

0x6a303d481d914b5b889611617f0066f3

>> cl work valid_worksheet_q6d4agn39u5uht8i18k9j8v335t98jqo
 (exit code 0, expected 0)

Switched to worksheet: http://rest-server:2900/worksheets/0x6a303d481d914b5b889611617f0066f3 (valid_worksheet_q6d4agn39u5uht8i18k9j8v335t98jqo)

>> cl wedit --tag=codalab-sample-worksheet --title=Other Worksheet q6d4agn39u5uht8i18k9j8v335t98jqo
 (exit code 0, expected 0)

Saved worksheet metadata for valid_worksheet_q6d4agn39u5uht8i18k9j8v335t98jqo(0x6a303d481d914b5b889611617f0066f3).

>> cl run echo codalab rules! --tags=codalab-sample-worksheet
 (exit code 0, expected 0)

0x1f25d8f28e204d90b6ad819ddae2f646

>> cl new valid_private_worksheet_q6d4agn39u5uht8i18k9j8v335t98jqo
 (exit code 0, expected 0)

0xcfc9cb5335f740a8817df5b573ea68b1

>> cl work valid_private_worksheet_q6d4agn39u5uht8i18k9j8v335t98jqo
 (exit code 0, expected 0)

Switched to worksheet: http://rest-server:2900/worksheets/0xcfc9cb5335f740a8817df5b573ea68b1 (valid_private_worksheet_q6d4agn39u5uht8i18k9j8v335t98jqo)

>> cl wedit --tag=codalab-sample-worksheet --title=Other Private Worksheet q6d4agn39u5uht8i18k9j8v335t98jqo
 (exit code 0, expected 0)

Saved worksheet metadata for valid_private_worksheet_q6d4agn39u5uht8i18k9j8v335t98jqo(0xcfc9cb5335f740a8817df5b573ea68b1).

>> cl wperm 0xcfc9cb5335f740a8817df5b573ea68b1 public none
 (exit code 0, expected 0)

Group public(0xc573c2c89326443a97e96edaf6443e51) has none permission on worksheet valid_private_worksheet_q6d4agn39u5uht8i18k9j8v335t98jqo(0xcfc9cb5335f740a8817df5b573ea68b1).

>> cl run echo private run --tags=codalab-sample-worksheet
 (exit code 0, expected 0)

0x53f7d206d7b349379be593789ef20e8b

>> cl perm 0x53f7d206d7b349379be593789ef20e8b public none
 (exit code 0, expected 0)

Group public(0xc573c2c89326443a97e96edaf6443e51) has none permission on 1 bundles.

>> cl search .png .limit=3 created=.sort- --uuid-only
 (exit code 0, expected 0)

0x33f2c04ec0104a91bb1b6635657f302a
0x024ed1e3cdb54dc28c257eea784f171a
0x40ecdb2b78df45fdb0fb7d51d5574b1a

>> cl search .html .limit=3 created=.sort- --uuid-only
 (exit code 0, expected 0)

0x1039cbd9dd2d4d3a88368e46c54b5f83
0x4817b1d8c2594ad4832c612cb0a2aeb9
0x78b33093f3be46a8b620e4e1a62f80db

>> cl search .tsv .limit=3 created=.sort- --uuid-only
 (exit code 0, expected 0)

0x9fe14d0e32214311b00d74b2cca93f66
0xee9303db2f9c436f84b1f90e947ca3f9
0x85350d5ec8ae4ee1b7158d7e81281671

>> cl new cl_small_worksheet
 (exit code 0, expected 0)

0x9cee6305d11a4e42ba9ed237d4ef341f

>> cl work cl_small_worksheet
 (exit code 0, expected 0)

Switched to worksheet: http://rest-server:2900/worksheets/0x9cee6305d11a4e42ba9ed237d4ef341f (cl_small_worksheet)

>> cl wedit --tag=codalab-sample-worksheet --title=Small Worksheet
 (exit code 0, expected 0)

Saved worksheet metadata for cl_small_worksheet(0x9cee6305d11a4e42ba9ed237d4ef341f).

>> cl wedit --file=/tmp/sample-worksheet-temp.txt
 (exit code 0, expected 0)

Saved worksheet items for cl_small_worksheet(0x9cee6305d11a4e42ba9ed237d4ef341f).

Deleted worksheet file at /tmp/sample-worksheet-temp.txt.
Done.
>> cl print cl_small_worksheet
 (exit code 0, expected 0)

### Worksheet: http://rest-server:2900::cl_small_worksheet(0x9cee6305d11a4e42ba9ed237d4ef341f)
### Title: Small Worksheet
### Tags: codalab-sample-worksheet
### Owner: codalab(0)
### Permissions: public(0xc573c2):read

## Introduction
This is the **small** sample worksheet.

## Worksheet References

#### Valid Worksheet References

[Worksheet valid_worksheet_6911p0iovbtdrlr6626kqmug4083fv8s(0x4768b82eaf7b40b088ae9dd8cbeb5d43)]
[Worksheet valid_worksheet_dg72kyu0q0a4hk9l9qbhpukflkae4nfs(0x49c42dde0ad74043881cbae26fd90882)]
[Worksheet valid_worksheet_q6d4agn39u5uht8i18k9j8v335t98jqo(0x6a303d481d914b5b889611617f0066f3)]

#### Private Worksheet References

[Worksheet valid_private_worksheet_6911p0iovbtdrlr6626kqmug4083fv8s(0x2d5005a85ca24e0096434cbd7bee59ba)]
[Worksheet valid_private_worksheet_dg72kyu0q0a4hk9l9qbhpukflkae4nfs(0xb2a212ef2a7145709e82267e7047f5d7)]
[Worksheet valid_private_worksheet_q6d4agn39u5uht8i18k9j8v335t98jqo(0xcfc9cb5335f740a8817df5b573ea68b1)]

## Bundle References

#### Valid Bundle Referen (...more...)
Finished validating content of the sample worksheet...
Success.
--- Completion Time: 0.9746117512385051 minutes---
```

teetone committed:
Merge branch 'master' of https://github.com/codalab/codalab-worksheets into 547
